src/sheet/sheet.c

```diff
@@ -317,6 +316,10 @@ gboolean sheet_get_adjustments (const Sheet *sheet, GtkAdjustment **hadj, GtkAdj
  */
 void sheet_change_zoom (Sheet *sheet, gdouble factor)
 {
+	double zoom;
+	sheet_get_zoom (sheet, &zoom);
+	sheet_set_zoom (sheet, zoom * factor);
+
```

Comment:
How was that wrong? This API was specifically there to zoom the sheet and abstract away from whatever it is internally (GooCanvas or whatever) so it can be used conveniently after construction.

If you want to set the zoom from within sheet without rescaling, use `sheet->priv->zoom = zoom;`